specs/language/lex.tex

```diff
@@ -0,0 +1,48 @@
+\Ch{Lexical Conventions}{Lex}
+
+\Sec{Unit of Translation}{Lex.Translation}
+
+\p The text of \acrshort{hlsl} programs is collected in \textit{source} and
+\textit{header} files. The distinction between source and header files is social
+and not technical. An implementation will construct a \textit{translation unit}
+from a single source file and any included source or header files referenced via
+the \texttt{\#include} preprocessing directive conforming to the \gls{isoC}
+preprocessor specification.
+
+\p An implementation may implicitly include additional sources as required to
+expose the \acrshort{hlsl} library functionality as defined in (FIXME: Add
+reference to library chapter).
+
+\Sec{Phases of Translation}{Lex.Phases}
+
+\p \acrshort{hlsl} inherits the phases of translation from \gls{isoCPP}, with
+minor alterations, specifically the removal of support for trigraph and digraph
+sequences. Below is a description of the phases.
+
+\begin{enumerate}
+  \item Source files are characters are mapped to the basic source character set
+  in an implementation-defined manner.
+  \item Any sequence of backslash (\texttt{\textbackslash}) immediately followed
+  by a new line is deleted, resulting in splicing lines together.
+  \item Tokenization occurs and comments are isolated. If a source file ends in
+  a partial comment or preprocessor token the program is ill-formed and a
+  diagnostic shall be issued. Each comment block shall be treated as a single
+  white-space character.
+  \item Preprocessing directives are executed, macros are expanded,
```

Comment:
We have a flag `flegacy-macro-expansion` that affects macro expansion that might bear mentioning?

Reply:
Given that is a legacy behavior, I don't think it describes the current HLSL language. My goal for this spec is to describe HLSL 2021 as we want the language to be rather than legacy support we might enable under flags or other compatibility modes.